### PR TITLE
feat(api): Feature variable APIs now return default variable value when featureEnabled property is false.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Feature variable APIs now return default variable value when featureEnabled property is false.

## Test plan
- Added unit tests.

### CODE REVIEW

File: OptimizelySDKCore/OptimizelySDKCore/Optimizely.m [hunk 1]
```diff
@@ -300,7 +300,7 @@ - (NSString *)getFeatureVariableValueForType:(NSString *)variableType
         OPTLYVariation *variation = decision.variation;
         OPTLYVariableUsage *featureVariableUsage = [variation getVariableUsageForVariableId:featureVariable.variableId];
         
-        if (featureVariableUsage) {
+        if (featureVariableUsage && variation.featureEnabled) {
```
Comment (aliabbasrizvi) [inline OptimizelySDKCore/OptimizelySDKCore/Optimizely.m]: Log message needs to be fixed as pointed out in other SDKs here as well. The new scenario is a situation where user is in variation, but has feature disabled and hence the default value.

**Review by aliabbasrizvi — DISMISSED** [review]

File: OptimizelySDKCore/OptimizelySDKCore/Optimizely.m (hunk 1 above)
Comment (yasirfolio3) [reply OptimizelySDKCore/OptimizelySDKCore/Optimizely.m]: Suggested changes made.

**Review by jaeopt — APPROVED** [review]
LGTM

**Comment by aliabbasrizvi** [comment]
@yasirfolio3 can you resolve the conflicts with master and send back for review?

**Comment by yasirfolio3** [comment]
Conflicts resolved, please review now.
@aliabbasrizvi

**Pull request merged**